src/Symfony/Component/Clock/MockClock.php

```diff
@@ -49,12 +64,12 @@ public function __construct(\DateTimeImmutable|string $now = 'now', \DateTimeZon
 
     public function now(): DatePoint
     {
-        return clone $this->now;
+        return clone $this->now ??= $this->tryFromTimeProvider();
```

Comment:
Should the `$now` property really be populated with result of the closure? What if I wanted to produce a different result each time `now()` is called?

Reply:
> Should the $now property really be populated with result of the closure?

Yes, otherwise it defeats the MockClock purpose to me

> What if I wanted to produce a different result each time now() is called?

The closure can take care of the return values, and `MockClock::reset()` should be called to indicate de closure should be executed again next time `$now` is queried

Reply:
The thing is I then won't still not be able to test a piece of code where the clock is called at least to times but where my test wants to verify some assertions of a certain being passed from the first to the second call. If the callable would be executed each time `now()` is called I could use a static counter in the closure to have an increase of time between the two calls.

Reply:
Also: since PSR's ClockInterface being a functional one (it has one method), we don't need any implementation to wire a closure as a cock. AutowireCallable FTW.

Reply:
I got your use case. Still, shouldn't we keep this clock like this with on-demand reloading? Your use case could be solved using some kind of `SpyClock` that would decorate the MockClock, then adding a counter when calling `now()`. This way, you would enjoy nice performances as well as a counter if you need it. What do you think of this approach?